linkerd/proxy/http/src/server.rs

```diff
@@ -28,24 +28,15 @@ pub struct NewServeHttp<F, H> {
     drain: drain::Watch,
 }
 
-/// Accepts HTTP connections.
-///
-/// The server accepts TCP connections with their detected protocol. If the
-/// protocol is known to be HTTP, a server is built with a new HTTP service
-/// (built using the `H`-typed NewService).
-///
-/// Otherwise, the `F` type forwarding service is used to handle the TCP
-/// connection.
-#[derive(Debug)]
-pub struct ServeHttp<T, F: NewService<T>, H: NewService<(HttpVersion, T)>> {
-    target: T,
-    new_tcp: F,
-    tcp: Option<F::Service>,
-    new_http: H,
-    http1: Option<H::Service>,
-    h2: Option<H::Service>,
-    server: hyper::server::conn::Http<trace::Executor>,
-    drain: drain::Watch,
+#[derive(Clone, Debug)]
+pub enum ServeHttp<F, H> {
+    Opaque(F, drain::Watch),
```

Comment:
nit: it seems a little weird to me to have something named `ServeHttp` that might be serving a protocol that isn't HTTP...but i notice that we already did that, so it's not a blocker for _this_ PR...